### DISCUSSION TITLE
Grafana v9.0.3 on IBM zLinux

### DISCUSSION THREAD

**Discussion by sinceresiva** [swc-project/swc#5266] [Q&A] [answered]
Hi,

Grafana v9.0.3 is using a version of @swc/core **(v1.2.160)** which was giving the "Unsupported Architecture" error in IBM z-Linux (s390x) machines. I have now changed the Grafana package.json to use the latest version of @swc/core **v1.2.218**. Now I get a different error "Bindings not found" as shown in the error in below screenshot (on executing: yarn start). From PR (#5233)  swc is supposed to interop between @swc/core and @swc/wasm. Could you pls give me any relevant configuration details to be included in package.json? Am I supposed to explicitly include @swc/wasm (v1.2.218) in package.json? Pls help.
<img width="479" alt="bindingsnotfound" src="https://user-images.githubusercontent.com/4971391/180232408-b2385695-9a87-415c-8fe1-d938a4f0b018.png">

My environment is:
OS: IBM z-Linux (s390x)
NodeJS version: v16.15.1
yarn version: 4.0.0-rc.12
npm version: 8.11.0

**Comment by kwonoj** [comment]
I'm seeing you're repeating same question, but there's really no way to provide any suggestion / help if you do not provide necessary details to debug such as proper installation log, error messages, NOT the runtime error message itself.

**Reply by sinceresiva** [reply]
I do apologize for not having uploaded the relevant logs/installation log. Please find below attached.
1. The install instructions of Grafana 9.0.3 is attached in GrafanaInstallInstructions.txt
2. The **yarn start** and **yarn build** logs are attached in their respective files. The **yarn build** error occurs after I manually include "@swc/wasm":1.2.218 in package.json. 
[GrafanaInstallInstructions.txt](https://github.com/swc-project/swc/files/9190913/GrafanaInstallInstructions.txt)
[YarnBuildError.txt](https://github.com/swc-project/swc/files/9190914/YarnBuildError.txt)
[YarnStartError.txt](https://github.com/swc-project/swc/files/9190915/YarnStartError.txt)

Please note that the above issue happens only in IBM zLinux (s390x). In Ubuntu (installed and accessed through Windows WSL) Grafana works just fine.

Thanks, Siva

**Reply by kwonoj** [reply]
Please note I explicitly requested **proper installation log**. Attached is nothing related to this issue.

**Reply by JavaPerformance** [reply]
Hi, I also added swc/wasm manually and got a little further...

bash-5.1# yarn start
RuntimeError: memory access out of bounds
    at <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_map::h3c249c9cff355d72 (wasm://wasm/0514c836:1:2510589)
    at wasm_bindgen::JsValue::into_serde::hb01d53b15dca0442 (wasm://wasm/0514c836:1:15791007)
    at transformSync (wasm://wasm/0514c836:1:4311733)
    at Object.module.exports.transformSync (/root/.yarn/berry/cache/@swc-wasm-npm-1.2.218-1ab1e15f9d-9.zip/node_modules/@swc/wasm/wasm.js:191:14)
    at Compiler.transformSync (/root/grafana/.yarn/unplugged/@swc-core-npm-1.2.218-ab87518b3f/node_modules/@swc/core/index.js:225:37)
    at Object.transformSync (/root/grafana/.yarn/unplugged/@swc-core-npm-1.2.218-ab87518b3f/node_modules/@swc/core/index.js:325:21)
    at Object.transpile (/root/grafana/.yarn/__virtual__/ts-node-virtual-c1e63b914e/2/.yarn/berry/cache/ts-node-npm-10.7.0-ef39b1d45e-9.zip/node_modules/ts-node/src/transpilers/swc.ts:148:39)
    at /root/grafana/.yarn/__virtual__/ts-node-virtual-c1e63b914e/2/.yarn/berry/cache/ts-node-npm-10.7.0-ef39b1d45e-9.zip/node_modules/ts-node/src/index.ts:1276:35
    at Object.compile (/root/grafana/.yarn/__virtual__/ts-node-virtual-c1e63b914e/2/.yarn/berry/cache/ts-node-npm-10.7.0-ef39b1d45e-9.zip/node_modules/ts-node/src/index.ts:1322:43)
    at Module.m._compile (/root/grafana/.yarn/__virtual__/ts-node-virtual-c1e63b914e/2/.yarn/berry/cache/ts-node-npm-10.7.0-ef39b1d45e-9.zip/node_modules/ts-node/src/index.ts:1454:30)

**Reply by kwonoj** [reply]
I believe

```
RuntimeError: memory access out of bounds
at <&mut serde_json:🇩🇪:Deserializer as serde:🇩🇪:Deserializer>::deserialize_map::
```

This is a separate issue that needs to be investigated further. Please file an issue with minimal codes can repro this.

**Comment by sinceresiva** [comment]
Yarn does not generate separate installation logs in the file system. I have attached both yarn install and yarn start logs that we get to see in the console.

**Reply by kwonoj** [reply]
No for both. First, yarn supports installation log (https://yarnpkg.com/cli/bin/#options-v%2C-verbose) Secondly, you did not attach any installation logs.

**Comment by kwonoj** [comment, marked as answer]
While there's some lacks of info for further debugging, `@swc/core` meanwhile got some updates to relieve additional pkg installation via https://github.com/swc-project/swc/pull/5322. It is not released yet, but once new `@swc/core` released it'll be applied.

Note this **does not mean** it resolves all of the problems. There are some known issues with fallback bindings, which supposed to be investigated further and it should be filed separately.

**Comment by JavaPerformance** [comment]
I finally got it to work... it's a bit hacky but acts as a guide.

https://github.com/JavaPerformance/GrafanaS390xBuild
